### PR TITLE
Fix potential double free in core_pkcs11.c

### PR DESCRIPTION
**Pull request by lundinc2**
Reported in https://github.com/FreeRTOS/corePKCS11/issues/87.

### CODE REVIEW

File: source/core_pkcs11.c [hunk 1]
```diff
@@ -134,6 +134,7 @@ CK_RV xGetSlotList( CK_SLOT_ID ** ppxSlotId,
     if( ( xResult != CKR_OK ) && ( pxSlotId != NULL ) )
     {
         PKCS11_FREE( pxSlotId );
+        *ppxSlotId = NULL;
```
Comment (ravibhagavandas) [inline source/core_pkcs11.c:137]: Could we add an assertion in test that the slot pointer returned is NULL.

Comment (lundinc2) [reply source/core_pkcs11.c:137]: I don't think it is possible to stub out functions from the same source file.

**Review by ravibhagavandas — APPROVED** [review]

**Review by gkwicker — APPROVED** [review]
